### PR TITLE
Set cmake PATH on macos to address libzstd flakiness

### PR DESCRIPTION
**Pull request by huydhn**
This is to address the recent flakiness issue on MacOS ARM64 https://hud.pytorch.org/failure/Library%20not%20loaded%3A%20%40rpath%2Flibzstd.1.dylib.  

From what I see, the immediate cause is that `cmake` exec under `/Users/ec2-user/runner/_work/_temp/miniconda/pkgs/cmake-3.22.1-hae769c0_0/bin/` is used instead of the expected one under the temp CONDA_ENV, i.e. `/Users/ec2-user/runner/_work/_temp/conda_environment_3736476178/bin`.  I'm not quite sure what is the reason behind this flaky behavior, so I want to try a catch-all fix by setting the cmake PATH correctly

This PR also prints some debugging information w.r.t cmake PATH, and cleans up some legacy code in `macos-test.sh` script.

### CODE REVIEW

File: .jenkins/pytorch/macos-test.sh [hunk 1]
```diff
@@ -4,22 +4,9 @@
 # shellcheck source=./macos-common.sh
 source "$(dirname "${BASH_SOURCE[0]}")/macos-common.sh"
 
-if [ -z "${CI}" ]; then
```
Comment (huydhn) [inline .jenkins/pytorch/macos-test.sh:7]: These are legacy code.  Looking at the CI log, none of these are run anymore (python3.6):

* x86-64: https://github.com/pytorch/pytorch/actions/runs/3735754301/jobs/6339606870
* arm64: https://github.com/pytorch/pytorch/actions/runs/3735754301/jobs/6339527686

File: .jenkins/pytorch/macos-test.sh [hunk 2]
```diff
@@ -4,22 +4,9 @@
...
-  rm -rf "${WORKSPACE_DIR}"/miniconda3/lib/python3.6/site-packages/torch*
-fi
-
-export CMAKE_PREFIX_PATH=${WORKSPACE_DIR}/miniconda3/
```
Comment (ZainRizvi) [inline .jenkins/pytorch/macos-test.sh:11]: Is this var def not needed? I wasn't guarded by the `if [ -z "${CI}" ]; then` like the other lines

**Review by ZainRizvi — APPROVED** [review]

File: .jenkins/pytorch/macos-test.sh (hunk 2 above)
Comment (huydhn) [reply .jenkins/pytorch/macos-test.sh:11]: Yeah, this is definitely not needed anymore.  In all MacOS jobs, i.e. https://github.com/pytorch/pytorch/actions/runs/3739211153/jobs/6346399018, it's assigned to `export CMAKE_PREFIX_PATH=/miniconda3/` which is actually an invalid path.  And thing still works

**Pull request closed without merging**